main.tf

```diff
@@ -118,11 +117,13 @@ resource "aws_s3_bucket_policy" "lambda_deploy" {
 
 # S3 object to hold the deployed artifact
 resource "aws_s3_bucket_object" "lambda_deploy_object" {
-  bucket = aws_s3_bucket.lambda_deploy.id
-  etag   = filemd5(var.path)
-  key    = local.deploy_artifact_key
-  source = var.path
-  tags   = var.tags
+  bucket      = data.aws_ssm_parameter.deployment_bucket_id.value
+  key         = "${var.name}/${local.deploy_artifact_key}"
+  source      = var.path
+  source_hash = md5(local.source_hash)
+  tags = merge(var.tags, {
+    GitSHA = var.git_sha
```

Comment:
I think this will be okay if the git sha happens to be null.
> The tag value must be a minimum of 0 and a maximum of 256 Unicode characters in UTF-8.